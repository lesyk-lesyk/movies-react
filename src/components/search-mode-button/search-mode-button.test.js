import React from 'react';
import { SearchModeButton } from './index';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
jest.mock('../button/button');

describe('SearchModeButton', () => {
  it('renders red button if mode equal to searchBy in store', () => {
    const store = configureStore()({ searchBy: 'title' });
    const tree = renderer
      .create(
        <SearchModeButton mode='title' store={store}>Title </SearchModeButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders gray button if mode not equal to searchBy in store', () => {
    const store = configureStore()({ searchBy: 'title' });
    const tree = renderer
      .create(
        <SearchModeButton mode='other' store={store}>Title </SearchModeButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
