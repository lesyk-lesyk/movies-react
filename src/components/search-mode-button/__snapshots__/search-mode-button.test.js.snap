// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModeButton renders gray button if mode not equal to searchBy in store 1`] = `
<Button
  className="search-mode-btn"
  color="gray"
  onClick={[Function]}
  size="sm"
>
  Title 
</Button>
`;

exports[`SearchModeButton renders red button if mode equal to searchBy in store 1`] = `
<Button
  className="search-mode-btn"
  color="red"
  onClick={[Function]}
  size="sm"
>
  Title 
</Button>
`;
